include/utils/tree.h

```diff
@@ -76,16 +76,19 @@ class Tree : public TreeBase
 
   /**
    * @returns a pointer to the element containing point p,
+   * for elements of dimension elem_dim,
    * optionally restricted to a set of allowed subdomains,
    * optionally using a non-zero relative tolerance for searches.
    */
-  const Elem* find_element(const Point& p,
-                           const std::set<subdomain_id_type>
-                           *allowed_subdomains = NULL,
-                           Real relative_tol = TOLERANCE) const;
```

Comment:
For crying out loud, even reading the diff I'm getting confused by all the find_element functions. I'll fix this up in a minute.